posthog/api/github.py

```diff
@@ -0,0 +1,162 @@
+            raise SignatureVerificationError("No public key found matching key identifier")
+
+        pem = entry.get("key")
+        if not isinstance(pem, str) or not pem.strip():
+            raise SignatureVerificationError("Malformed public key entry")
+
+        cache.set(cache_key, pem, TWENTY_FOUR_HOURS)
+
+    try:
+        pub = serialization.load_pem_public_key(pem.encode("utf-8"))
+    except Exception as e:
+        raise SignatureVerificationError("Unable to parse public key") from e
+
+    if not isinstance(pub, ec.EllipticCurvePublicKey) or pub.curve.name.lower() not in ("secp256r1", "prime256v1"):
+        raise SignatureVerificationError("Unsupported public key type/curve (expected ECDSA P-256)")
+
+    try:
+        sig_bytes = base64.b64decode(sig, validate=True)
+    except Exception as e:
+        raise SignatureVerificationError("Signature is not valid base64") from e
+
+    message = payload.encode("utf-8")
+
+    try:
+        pub.verify(sig_bytes, message, ec.ECDSA(hashes.SHA256()))
+    except InvalidSignature as e:
+        raise SignatureVerificationError("Signature does not match payload") from e
+
+
+class SecretAlertSerializer(serializers.Serializer):
+    token = serializers.CharField()
+    type = serializers.ChoiceField(choices=["posthog_personal_api_key", "posthog_feature_flags_secure_api_key"])
+    url = serializers.CharField(allow_blank=True)
+    source: Any = serializers.CharField()
+
+
+class SecretAlert(APIView):
+    authentication_classes = []
+    permission_classes = [AllowAny]
+    parser_classes = [JSONParser]
+
+    def post(self, request):
+        kid = (request.headers.get("Github-Public-Key-Identifier") or "").strip()
+        sig = (request.headers.get("Github-Public-Key-Signature") or "").strip()
+
+        if not kid:
+            raise ValidationError(
+                {
+                    "headers": {
+                        "Github-Public-Key-Identifier": "required non-blank string",
+                    }
+                }
+            )
+        if not sig:
+            raise ValidationError(
+                {
+                    "headers": {
+                        "Github-Public-Key-Signature": "required non-blank string",
+                    }
+                }
+            )
+
+        try:
+            verify_github_signature(request.body.decode("utf-8"), kid, sig)
+        except SignatureVerificationError:
+            return Response({"detail": "Invalid signature"}, status=401)
+
+        if not isinstance(request.data, list):
+            raise ValidationError(detail="Expected a JSON array")
+        if len(request.data) < 1:
+            raise ValidationError(detail="Array must contain at least one item")
+
+        secret_alert = SecretAlertSerializer(data=request.data, many=True)
+        secret_alert.is_valid(raise_exception=True)
+        items = secret_alert.validated_data
+
+        results = []
+        for _, item in enumerate(items):
```

Comment:
If you're not interested in the index, why use `enumerate`?
```suggestion
        for item in items:
```

Reply:
Because I'm a total python noob :)